src/main/java/com/nylas/EventQuery.java

```diff
@@ -111,5 +131,20 @@ public EventQuery endsAfter(Instant endsAfter) {
 		this.endsAfter = endsAfter;
 		return this;
 	}
+
+	public EventQuery metadataKey(String... metadataKey) {
+		this.metadataKey = metadataKey;
+		return this;
+	}
+
+	public EventQuery metadataValue(String... metadataValue) {
+		this.metadataValue = metadataValue;
+		return this;
+	}
+
+	public EventQuery metadataPair(Map<String, String> metadataPair) {
+		this.metadataPair = metadataPair;
+		return this;
+	}
```

Comment:
Since we ended up working concurrently on this support, I have another implementation that I would prefer if you don't mind.  It looks like this:

Fields:
```	private List<String> metadataKeys;
	private List<String> metadataValues;
	private List<String> metadataPairs;
```

addParameters:
```
		if (metadataKeys != null) {
			for (String key : metadataKeys) {
				url.addQueryParameter("metadata_key", key);
			}
		}
		if (metadataValues != null) {
			for (String value : metadataValues) {
				url.addQueryParameter("metadata_value", value);
			}
		}
		if (metadataPairs != null) {
			for (String pair : metadataPairs) {
				url.addQueryParameter("metadata_pair", pair);
			}
		}
```

fluent filter methods:
```
	/**
	 * Add a required metadata key to the query.
	 * Returned events must include a metadata property with this key.
	 * This method can be invoked multiple times to require the event to have multiple
	 * metadata keys.  Each time this is called it adds another required key rather than
	 * replacing previous required keys.
	 */
	public EventQuery metadataKey(String metadataKey) {
		if (this.metadataKeys == null) {
			this.metadataKeys = new ArrayList<>();
		}
		this.metadataKeys.add(metadataKey);
		return this;
	}	
	
	/**
	 * Add a required metadata value to the query.
	 * Returned events must include a metadata property with this value.
	 * This method can be invoked multiple times to require the event to have multiple
	 * metadata values.  Each time this is called it adds another required value rather than
	 * replacing previous required values.
	 */
	public EventQuery metadataValue(String metadataValue) {
		if (this.metadataValues == null) {
			this.metadataValues = new ArrayList<>();
		}
		this.metadataValues.add(metadataValue);
		return this;
	}
	
	/**
	 * Add a required metadata key/value pair to the query.
	 * Returned events must include a metadata property with this key and value.
	 * This method can be invoked multiple times to require the event to have multiple
	 * metadata key value pairs.  Each time this is called it adds another required key value pair rather than
	 * replacing previous required key value pairs.
	 */
	public EventQuery metadataKeyValue(String key, String value) {
		if (this.metadataPairs == null) {
			this.metadataPairs = new ArrayList<>();
		}
		this.metadataPairs.add(key + ":" + value);
		return this;
	}
```

I prefer this style of fluent methods as it allows adding a single filter at a time rather than replacing all of them.

Reply:
Changes implemented!